### PR TITLE
Bach improve column name escaping

### PR DESCRIPTION
**Pull request by thijs-obj**
Changed the way we encode series-names as column-names, when the series name is not a valid column name.
The resulting column names contain more of the original characters, but they are not reversible anymore. The exact method can be found by inspecting `get_sql_column_name()`

### CODE REVIEW

File: bach/tests/unit/bach/test_utils.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def test_get_sql_column_name(dialect):
                        'test'),
         ColNameEncoded('Test',
                        'Test',
-                       '__esc_krsxg5a',
-                       '__esc_krsxg5a'),
+                       'test__bs6gmepvkq',
+                       'test__bs6gmepvkq'),
```
Comment (thijs-obj) [inline bach/tests/unit/bach/test_utils.py:75]: The difference in naming is visible here

File: bach/tests/functional/bach/test_revert_merge.py [hunk 2]
```diff
@@ -148,12 +165,12 @@ def test_revert_merge_grouped(engine):
 
 
 def _compare_source_with_replicate(original_source: DataFrame, replicate: DataFrame) -> None:
-    assert original_source.base_node == replicate.base_node
-    assert original_source.data_columns == replicate.data_columns
-    assert original_source.index == replicate.index
+    assert replicate.base_node == original_source.base_node
+    assert replicate.data_columns == original_source.data_columns
+    assert replicate.index == original_source.index
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_revert_merge.py:170]: This is not really related. But this improves the pytest error message. 'actual' and 'expected' are now the correct values

File: bach/tests/functional/bach/test_revert_merge.py [hunk 3]
```diff
@@ -18,6 +18,23 @@ def test_revert_merge_basic(engine):
...
+    bt = get_df_with_test_data(engine, full_data_set=False)[['skating_order', 'city']]
+    bt['city'] = bt['city'] + 'test'
+    bt = bt.rename(columns={'city': 'C1TŸ'})
+    bt = bt.materialize('https://github.com/objectiv/objectiv-analytics/issues/1430')
+    mt = get_df_with_food_data(engine)[['skating_order', 'food']]
+    mt = mt.rename(columns={'food': 'FÖÔD'})
+    mt = mt.materialize('https://github.com/objectiv/objectiv-analytics/issues/1430')
+    merged = bt.merge(mt)
+
+    print(mt.view_sql())
```
Comment (jansentom) [inline bach/tests/functional/bach/test_revert_merge.py:31]:
```suggestion
```

File: bach/bach/utils.py [hunk 4]
```diff
@@ -108,55 +113,92 @@ def get_sql_column_name(dialect: Dialect, name: str) -> str:
...
         raise ValueError(f'Column name "{name}" is not valid for SQL dialect {dialect.name}, and cannot be '
                          f'escaped. Try making the series name shorter and/or using characters [a-z] only.')
     return escaped
 
 
+def _get_string_hash(input: str) -> str:
+    """ Give md5 hash of the given input strings, encoded as a 26 character long base-32 string. """
+    # .hexdigest() would give the md5 hash encoded with [a-e0-9], which gives 4 bits per character
+    # we'd rather not spend too many characters on the hash, so we base-32 encode the bytes, which gives 5
+    # bits per character. Which makes the 128-bit hash fit in 26 characters instead of 32.
```
Comment (jansentom) [inline bach/bach/utils.py:188]: nice touch

File: bach/bach/utils.py [hunk 5]
```diff
@@ -108,55 +113,92 @@ def get_sql_column_name(dialect: Dialect, name: str) -> str:
...
+    # This process should yield a valid column name for any database, if it's not too long.
+
+    table_upper_to_lower = str.maketrans(string.ascii_uppercase, string.ascii_lowercase)
+    chars_lower = set(string.ascii_lowercase)
+    chars_numbers = set('0123456789')
+    chars_underscore = set('_')
+    allowed_chars = chars_lower | chars_numbers | chars_underscore
+
+    # Convert 'name' into cleaned form, in three steps:
+    # 1) Normalize the name to the decomposed character form. This will split diacritics of the base
```
Comment (jansentom) [inline bach/bach/utils.py:147]: very nice flow and well explained, thnx

**Review by jansentom — APPROVED** [review]
A-ok, very nice!

File: bach/bach/utils.py (hunk 5 above)
Comment (KathiaBarahona) [reply bach/bach/utils.py:147]: :heavy_plus_sign:

File: bach/tests/functional/bach/test_revert_merge.py (hunk 3 above)
Comment (thijs-obj) [reply bach/tests/functional/bach/test_revert_merge.py]: doh! sharp

Fixed in 391af415091ebb026a26dd46d017c109bf2fae26
